src/services/services-manager.ts

```diff
@@ -16,47 +38,156 @@ const SERVICE_NAME_MAP: { [key in ServiceType]: string } = {
   data: "data"
 };
 
-const SERVICE_POST_INFO_MAP: {
+export const SERVICE_POST_INFO_MAP: {
   [key in ServiceType]: (qViewer: QueryViewer) => any;
 } = {
   metadata: metaData,
   data: data
 };
 
+export type ServicesContext = {
+  UseGXquery: boolean;
+  BaseUrl: string;
+  Generator: GeneratorType;
+  MetadataName: string;
+  ObjectName: string;
+  SerializedObject: string;
+};
+
 /**
  * Necessary so that the explorer does not cache the results of the services
  * @returns Unique value with each call, using the current date
  */
 const foolCache = () => new Date().getTime();
 
+const QueryToQueryProperties = (query: Query): QueryViewerServiceProperties => {
+  return {
+    Type: QueryViewerOutputType[
```

Comment:
The `as` operator should not be used, because it is not necessary. The problem with the type inference in this case lies in the type of the `query` parameter. The parameter's type is `any`, which is not good for inference. It must be correctly typed.